core/services/job/orm.go

```diff
@@ -364,6 +364,10 @@ func (o *orm) InsertJob(job *Job, qopts ...pg.QOpt) error {
 // DeleteJob removes a job
 func (o *orm) DeleteJob(id int32, qopts ...pg.QOpt) error {
 	o.lggr.Debugw("Deleting job", "jobID", id)
+	// Added a 1 minute timeout to this query since this can take a long time as data increases.
```

Comment:
Out of curiosity, if this query starts timing out, then who/what eventually ends up deleting such jobs?
Is there a reaper or similar to delete the old/obsolete jobs?

Reply:
Jobs are deleted manually by NOPs. So there's no reaper to automatically delete this jobs that were attempted to be deleted. Since NOPs are the ones requesting the delete action, they just try to delete again if it times out or fails.

Reply:
But if a query times-out due to a lot of jobs in the Node, it likely will keep timing out in future too.

How would the nops get their node back in a good state.

@samsondav Is it right to say that with growing list of jobs on a node, they can reach a situation where the delete job operation could repeatedly time out, and block deletions?
Instead of putting a timeout, could we break the query into multiple smaller ones, so atleast they are able to pass?

Reply:
@prashantkumar1982 not really, because the number of jobs would have to be astronomical (no node operator is likely to be running 100k jobs any time soon).

The reason this delete might take a long time is due to lots of associated job runs (fk) being deleted and there is a reaper that takes care of that, which Vyzaldy has fixed.

Reply:
Thanks, got it.